pyiron_base/jobs/job/extension/server/queuestatus.py

```diff
@@ -226,7 +227,12 @@ def wait_for_job(job, interval_in_s=5, max_iterations=100):
                 if job.status.string in job_status_finished_lst:
                     finished = True
                     break
-                time.sleep(interval_in_s)
+                elif isinstance(job.server.future, Future):
+                    finished = True
+                    job.server.future.result(timeout=interval_in_s)
```

Comment:
```suggestion
                    job.server.future.result(timeout=interval_in_s)
                    finished = job.server.future.done()
```
Something to that effect has to there, otherwise it will break the loop early, no?

Sorry, I wasn't more explicit earlier.  I was in a rush and you see where it got.  It is the last thing that I see for now.